rocky/rocky/scheduler.py

```diff
@@ -196,30 +198,43 @@ def list_tasks(
         **kwargs,
     ) -> PaginatedTasksResponse:
         kwargs = {k: v for k, v in kwargs.items() if v is not None}  # filter Nones from kwargs
-        res = self._client.get("/tasks", params=kwargs)
+        endpoint = "/tasks"
+        logger.debug("Making request to %s with args %s", endpoint, kwargs)
+        # Endpoint expects a str.
+        res = self._client.post(endpoint, params=kwargs, data=json.dumps(kwargs.get("filters", None)))  # type: ignore
+        if res.status_code != httpx.codes.OK:
+            logger.warning(
+                "Unexpected response %s from endpoint %s with args %s which returns code %s.",
+                res.content,
+                endpoint,
+                kwargs,
+                res.status_code,
+            )
```

Comment:
You might be better of using `res.raise_for_status()`, and if necessary, catch it and produce a nice user error or throw a `SchedulerError`, or a specific scheduler error (e.g. implement `class TaskFilterError(SchedulerError): ...`)

The logged warning will not be seen by the user, while the error could've definitely be a user (input) error